pkg/httpclient/client.go

```diff
@@ -46,10 +85,18 @@ func (c *Client) Post(path string, contentType string, body io.Reader) (*http.Re
 	return c.Do(req)
 }
 
+// NoTimeout disables the timeout for an HTTP request.
+func NoTimeout() RequestOption {
+	return func(req *http.Request) {
+		ctx := context.WithValue(req.Context(), keyNoTimeout, true)
```

Comment:
nit: I probably would have used `struct{}{}` instead of `true` since a value is not actually needed here